### PR TITLE
[CatalogPromotions][Validation] add validation to catalog promotion, …

### PR DESCRIPTION
**Pull request by SirDomin**
…unify not number and empty amount

| Q               | A
| --------------- | -----
| Branch?         | master
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | 
| License         | MIT

image
image

### CODE REVIEW

File: src/Sylius/Bundle/PromotionBundle/Resources/translations/validators.en.yml [hunk 1]
```diff
@@ -15,7 +15,7 @@ sylius:
         invalid_type: Catalog promotion action type is invalid. Please choose a valid type.
         percentage_discount:
             not_in_range: The percentage discount amount must be between 0% and 100%.
-            not_number: The percentage discount amount must be a number.
+            not_number_empty: The percentage discount amount must be a number and can not be empty.
```
Comment (AdamKasp) [inline src/Sylius/Bundle/PromotionBundle/Resources/translations/validators.en.yml:18]:
```suggestion
            not_number_or_empty: The percentage discount amount must be a number and can not be empty.
```

File: src/Sylius/Bundle/PromotionBundle/Validator/Constraints/CatalogPromotionAction.php [hunk 2]
```diff
@@ -23,7 +23,7 @@ final class CatalogPromotionAction extends Constraint
 
     public string $notInRangeDiscount = 'sylius.catalog_promotion_action.percentage_discount.not_in_range';
 
-    public string $notNumber = 'sylius.catalog_promotion_action.percentage_discount.not_number';
+    public string $notNumber = 'sylius.catalog_promotion_action.percentage_discount.not_number_empty';
```
Comment (AdamKasp) [inline src/Sylius/Bundle/PromotionBundle/Validator/Constraints/CatalogPromotionAction.php:26]:
```suggestion
    public string $notNumberOrEmpty = 'sylius.catalog_promotion_action.percentage_discount.not_number_empty';
```

File: src/Sylius/Behat/Context/Ui/Admin/ManagingCatalogPromotionsContext.php [hunk 3]
```diff
@@ -215,6 +215,39 @@ public function iAddForVariantsRuleWithoutVariantsConfigured(): void
...
+     */
+    public function iAddPercentageDiscountActionWithoutAmountConfigured(): void
+    {
+        $this->formElement->addAction();
+    }
+
+    /**
+     * @When I add invalid percentage discount action with non number in amount
+     */
+    public function iAddInvalidPercentageDiscountActionWithNonNumberInAmount()
```
Comment (Zales0123) [inline src/Sylius/Behat/Context/Ui/Admin/ManagingCatalogPromotionsContext.php:229]: `: void`

File: src/Sylius/Behat/Context/Ui/Admin/ManagingCatalogPromotionsContext.php [hunk 4]
```diff
@@ -215,6 +215,39 @@ public function iAddForVariantsRuleWithoutVariantsConfigured(): void
...
+    {
+        $this->formElement->addAction();
+        $this->formElement->specifyLastActionDiscount('alot');
+    }
+
+    /**
+     * @Then I should be notified that a discount amount should be between 0% and 100%
+     */
+    public function iShouldBeNotifiedThatADiscountAmountShouldBeBetweenAnd(): void
```
Comment (Zales0123) [inline src/Sylius/Behat/Context/Ui/Admin/ManagingCatalogPromotionsContext.php:238]: `iShouldBeNotifiedThatADiscountAmountShouldBeBetween0And100Percent`?

File: src/Sylius/Behat/Context/Ui/Admin/ManagingCatalogPromotionsContext.php [hunk 5]
```diff
@@ -215,6 +215,39 @@ public function iAddForVariantsRuleWithoutVariantsConfigured(): void
...
+    {
+        Assert::same($this->formElement->getValidationMessageForAction(), 'The percentage discount amount must be between 0% and 100%.');
+    }
+
+    /**
+     * @Then I should be notified that a discount amount should be a number and can not be empty
+     */
+    public function iShouldBeNotifiedThatADiscountAmountShouldBeANumber()
```
Comment (Zales0123) [inline src/Sylius/Behat/Context/Ui/Admin/ManagingCatalogPromotionsContext.php:246]: `: void` 💃

File: src/Sylius/Bundle/PromotionBundle/Form/Type/CatalogAction/PercentageDiscountActionConfigurationType.php [hunk 6]
```diff
@@ -15,6 +16,9 @@ public function buildForm(FormBuilderInterface $builder, array $options): void
         $builder
             ->add('amount', PercentType::class, [
                 'label' => 'sylius.ui.amount',
+                'constraints' => [
+                    new NotBlank(['groups' => ['sylius']])
```
Comment (Zales0123) [inline src/Sylius/Bundle/PromotionBundle/Form/Type/CatalogAction/PercentageDiscountActionConfigurationType.php:20]: Missing coma at the end of file

File: src/Sylius/Behat/Context/Ui/Admin/ManagingCatalogPromotionsContext.php [hunk 7]
```diff
@@ -215,6 +215,39 @@ public function iAddForVariantsRuleWithoutVariantsConfigured(): void
...
+     */
+    public function iShouldBeNotifiedThatADiscountAmountShouldBeBetween0And100Percent(): void
+    {
+        Assert::same($this->formElement->getValidationMessageForAction(), 'The percentage discount amount must be between 0% and 100%.');
+    }
+
+    /**
+     * @Then I should be notified that a discount amount should be a number and cannot be empty
+     */
+    public function iShouldBeNotifiedThatADiscountAmountShouldBeANumber(): void
```
Comment (arti0090) [inline src/Sylius/Behat/Context/Ui/Admin/ManagingCatalogPromotionsContext.php:246]:
```suggestion
    public function iShouldBeNotifiedThatADiscountAmountShouldBeANumberAndCannotBeEmpty(): void
```

**Review by arti0090 — APPROVED** [review]

**Review by GSadee — APPROVED** [review]

**Review by AdamKasp — APPROVED** [review]

**Comment by GSadee** [comment]
Thank you, @SirDomin! :tada:

**Pull request merged**